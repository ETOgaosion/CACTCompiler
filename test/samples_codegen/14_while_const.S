iobuff:
	.zero 80
.text
.align 1
.globl	main
.type	main, @function
main:
	sd	ra, -8(sp)
	addi	t0, zero, 4
	sw	t0, -20(sp)
	mv	t0, zero
	sw	t0, -16(sp)
	mv	t0, zero
	sw	t0, -12(sp)
	addi	sp, sp, -40
	call	get_int
	addi	sp, sp, 40
	sw	a0, -28(sp)
	lw	t0, -28(sp)
	sw	t0, -12(sp)
L_1_main:
	lw	t0, -16(sp)
	lw	t1, -20(sp)
	slt	t2, t0, t1
	sb	t2, -32(sp)
	lb	t0, -32(sp)
	beqz	t0, L_0_main
	lw	t1, -16(sp)
	addi	t0, t1, 1
	sw	t0, -36(sp)
	lw	t0, -36(sp)
	sw	t0, -16(sp)
	lw	t1, -12(sp)
	addi	t0, t1, 1
	sw	t0, -24(sp)
	lw	t0, -24(sp)
	sw	t0, -12(sp)
	lw	a0, -12(sp)
	addi	sp, sp, -40
	call	print_int
	addi	sp, sp, 40
	lla	t0, L_1_main
	jr	t0
L_0_main:
	mv	a0, zero
	ld	ra, -8(sp)
	ret
