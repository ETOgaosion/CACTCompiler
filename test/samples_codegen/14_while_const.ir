============ global var ================
============ imm value ================
============ functions ================
----------- funcDef -----------
main() => 2:
----------- paramVars -----------
----------- localVars -----------
b := symbol type: 1; data type: 2; initValue: 0

i := symbol type: 1; data type: 2; initValue: 0

n := symbol type: 2; data type: 2; initValue: 4

----------- tempVars ------------
t_5:= symbol type: TEMP VAR; data type: 2
t_0:= symbol type: TEMP VAR; data type: 2
t_2:= symbol type: TEMP VAR; data type: 1
t_3:= symbol type: TEMP VAR; data type: 2
----------- Codes ------------
main:
	n = 4;
	i = 0;
	b = 0;
	call get_int; 
	t_0 =  return ;
	t_1 <= b;
	t_1 = t_0;
L_1_main:
	t_2 = i < n
	beqz t_2 L_0_main
	t_3 = i + 1;
	t_4 <= i;
	t_4 = t_3;
	t_5 = t_1 + 1;
	t_6 <= b;
	t_6 = t_5;
	addparam t_6;
	call print_int; 
	goto L_1_main
L_0_main:
	return 0;

