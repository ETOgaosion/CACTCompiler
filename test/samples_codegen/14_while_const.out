bbl loader
please enter an int number:
7
8
9
10
